### PR TITLE
Results Ack Sending

### PR DESCRIPTION
**Pull request by Loonride**

### CODE REVIEW

File: funcx_forwarder/forwarder.py [hunk 1]
```diff
@@ -494,11 +501,32 @@ def handle_results(self):
...
+                self.handle_results_ack(endpoint_id, task.task_id)
+
         except zmq.Again:
             pass
         except Exception:
             logger.exception("Caught exception from results queue")
 
+    def handle_results_ack(self, endpoint_id, task_id):
+        if endpoint_id not in self.connected_endpoints:
+            return
```
Comment (yadudoc) [inline funcx_forwarder/forwarder.py:514]: A warning log line could be useful here if this check is needed at all. Meaning why are we receiving/processing a result from a not connect ep.

File: funcx_forwarder/forwarder.py [hunk 2]
```diff
@@ -494,11 +501,32 @@ def handle_results(self):
...
+            "endpoint_id": endpoint_id,
+            "task_id": task_id
+        })
+        try:
+            # send an ack
+            self.tasks_q.put(endpoint_id.encode('utf-8'),
+                             msg.pack())
+        except (zmq.error.ZMQError, zmq.Again):
+            logger.exception(f"Endpoint:{endpoint_id} results ack send failed")
+            self.disconnect_endpoint(endpoint_id)
```
Comment (yadudoc) [inline funcx_forwarder/forwarder.py:528]: Would be good to make sure that `self.disconnect_endpoint` is safe to call multiple times.

**Review by yadudoc — APPROVED** [review]
I suspect that we might be adding a few redundant logs, but lets check that separately. Changes here look good.

**Comment by Loonride** [comment]
[ch10083]

**Pull request merged**